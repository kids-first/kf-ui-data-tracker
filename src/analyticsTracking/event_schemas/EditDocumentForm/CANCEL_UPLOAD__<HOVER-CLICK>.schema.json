{
  "$id": "https://kf-ui-datatracker.kidsfirstdrc.org/event_schemas/CANCEL_UPLOAD/CANCEL_UPLOAD__<HOVER-CLICK>.schema.json",
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "CANCEL_UPLOAD__<HOVER-CLICK>",
  "type": "object",
  "description": "onMouseOver event for the 'Cancel' button in the EditDocumentForm component of the NewDocumentView",
  "allOf": [
    {"$ref": "../common_defintions.json#/path"},
    {"$ref": "../common_defintions.json#/route"},
    {"$ref": "../common_defintions.json#/view"},
    {"$ref": "../BUTTON/button_defintions.json#/buton_text"},
    {"$ref": "../BUTTON/button_defintions.json#/buton_type"},
    {"$ref": "../BUTTON/button_defintions.json#/link"}
  ]
}
