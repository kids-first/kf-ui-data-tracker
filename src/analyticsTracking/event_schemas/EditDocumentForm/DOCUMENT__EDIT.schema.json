{
  "$id": "https://kf-ui-datatracker.kidsfirstdrc.org/event_schemas/DOCUMENT/DOCUMENT__EDIT.schema.json",
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "DOCUMENT__EDIT",
  "type": "object",
  "description": "",
  "allOf": [
    {"$ref": "../common_defintions.json#/path"},
    {"$ref": "../common_defintions.json#/route"},
    {"$ref": "../common_defintions.json#/view"},
    {
      "properties": {
        "updated_version": {
          "type": "object",
          "properties": {
            "kfId": {
              "type": "string",
              "description": "kids first study file version identifier",
              "pattern": "^SF_[A-HJ-KM-NP-TV-Z0-9]{8}"
            },
            "file_name": {
              "type": "string",
              "description": "name of the local file associated with the file version"
            },
            "description": {
              "type": "string",
              "description": "user provided study document description field content"
            },
            "file_status": {
              "type": "string",
              "enum": ["PEN", "APP", "CHN", "PRC", "UPD"],
              "description": "Approval Status code"
            }
          }
        }
      }
    }
  ]
}
