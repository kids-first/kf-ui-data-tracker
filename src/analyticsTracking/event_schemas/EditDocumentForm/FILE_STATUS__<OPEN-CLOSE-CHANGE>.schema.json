{
  "$id": "https://kf-ui-datatracker.kidsfirstdrc.org/event_schemas/FILE_STATUS/FILE_STATUS__<OPEN-CLOSE-CHANGE>.schema.json",
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "FILE_STATUS__<OPEN-CLOSE-CHANGE>",
  "type": "object",
  "description": "onOpen, onClose, and onChange event schema for the `File Status` dropdown in the EditDocumentForm",
  "allOf": [
    {"$ref": "../common_defintions.json#/path"},
    {"$ref": "../common_defintions.json#/route"},
    {"$ref": "../common_defintions.json#/view"},
    {
      "properties": {
        "placeholder": {
          "type": "string",
          "description": "HTML placeholder value for the input element"
        },
        "value": {
          "type": "string",
          "description": "HTML input value at time of event, for onChange it is the value that the user selected"
        }
      }
    }
  ]
}
