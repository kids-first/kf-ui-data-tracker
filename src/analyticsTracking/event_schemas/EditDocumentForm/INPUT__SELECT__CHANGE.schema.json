{
  "$id": "https://kf-ui-datatracker.kidsfirstdrc.org/event_schemas/INPUT/INPUT__SELECT__CHANGE.schema.json",
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "INPUT__SELECT__CHANGE",
  "type": "object",
  "description": "onChange event for the SelectElement component",
  "allOf": [
    {"$ref": "../common_defintions.json#/path"},
    {"$ref": "../common_defintions.json#/route"},
    {"$ref": "../common_defintions.json#/view"},
    {"$ref": "../common_defintions.json#/study"},
    {"$ref": "../common_defintions.json#/file"},
    {
      "properties": {
        "value": {
          "type": "string",
          "description": "value set for the input element after change"
        },
        "input_type": {
          "type": "string",
          "description": "HTML type attibute for the input element"
        },
        "name": {
          "type": "string",
          "description": "name of the input value the element mutates"
        },
        "label": {
          "type": "string",
          "description": "HTML id attribute value for the input element"
        },
        "radio_text": {
          "type": "string",
          "description": "text show to user for the radio button value"
        }
      }
    }
  ]
}
