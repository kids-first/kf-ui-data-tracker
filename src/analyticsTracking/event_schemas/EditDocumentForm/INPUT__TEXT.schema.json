{
  "$id": "https://kf-ui-datatracker.kidsfirstdrc.org/event_schemas/INPUT/INPUT__TEXT.schema.json",
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "INPUT__TEXT",
  "type": "object",
  "description": "onChange event for input elements in the EditDocumentForm component ",
  "allOf": [
    {"$ref": "../common_defintions.json#/path"},
    {"$ref": "../common_defintions.json#/route"},
    {"$ref": "../common_defintions.json#/view"},
    {"$ref": "../common_defintions.json#/study"},
    {"$ref": "../common_defintions.json#/file"},
    {
      "properties": {
        "input_type": {
          "type": "string",
          "description": "HTML type attibute for the input element"
        },
        "name": {"type": "string", "enum": ["file_name", "file_description"]},
        "value": {"type": "string", "description": "user input"},
        "placeholder": {
          "type": "string",
          "description": "placeholder text used for the input element"
        },
        "validation": {
          "type": "object",
          "description": "validation object returned by Formik, this is used to see if there are any string similarities in the document name"
        }
      }
    }
  ]
}
