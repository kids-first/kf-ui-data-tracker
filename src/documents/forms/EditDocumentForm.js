/* eslint-disable  no-useless-escape */
import React, {useState} from 'react';
import PropTypes from 'prop-types';
import {Form, Dropdown, Segment} from 'semantic-ui-react';
import {Formik, Field} from 'formik';
import validate from './validate';
import ExistingDocsMessage from './ExistingDocsMessage';
import SelectElement from '../components/FileDetail/SelectElement';
import Badge from '../../components/Badge/Badge';
import UploadWizard from '../modals/UploadWizard/UploadWizard';
import {fileTypeDetail, versionState} from '../../common/enums';
import MarkdownEditor from '../components/FileDetail/MarkdownEditor';
import draftToMarkdown from 'draftjs-to-markdown';
import {EditorState, convertToRaw, ContentState} from 'draft-js';
/**
 * Form to edit document information or annotate new file for uploading
 */
const EditDocumentForm = React.forwardRef(
  (
    {
      fileNode,
      fileType,
      fileName,
      fileDescription,
      versionStatus,
      history,
      handleSubmit,
      submitButtons,
      studyFiles,
    },
    ref,
  ) => {
    const [onUploading, setUploading] = useState(false);
    const [showDialog, setShowDialog] = useState(false);
    const [titleError, setTitleError] = useState({});
    const [editorState, setEditorState] = useState(
      EditorState.createWithContent(
        ContentState.createFromText(fileDescription || ''),
      ),
    );
    const options = ['PEN', 'APP', 'CHN', 'PRC'].map(state => ({
      key: state,
      value: state,
      text: versionState[state].title,
      content: <Badge state={state} />,
    }));
    return (
      <>
        <Formik
          initialValues={{
            file_name: fileName || '',
            file_type: fileType,
            file_desc: fileDescription,
            file_status: versionStatus,
          }}
          validate={vals => {
            setTitleError(validate(vals, fileNode, studyFiles));
            let errors = {};
            ['file_name', 'file_type'].forEach(function(field) {
              if (!vals[field]) {
                errors[field] = 'Required';
              }
            });
            const rawState = convertToRaw(editorState.getCurrentContent());
            const mdText = draftToMarkdown(rawState);
            if (mdText.length < 3) {
              errors.file_desc = 'Required';
            }
            return errors;
          }}
          onSubmit={values => {
            setUploading(true);
            const rawState = convertToRaw(editorState.getCurrentContent());
            const mdText = draftToMarkdown(rawState);
            var submitValue = values;
            submitValue.file_desc = mdText;
            handleSubmit(...Object.values(values));
          }}
        >
          {({
            values,
            handleChange,
            handleBlur,
            handleSubmit,
            setFieldValue,
            errors,
            touched,
            validateForm,
          }) => (
            <>
              {titleError.file_name &&
                (titleError.file_name.existing_similarity ||
                  titleError.file_name.exact_matches ||
                  titleError.file_name.upload_similarity) && (
                  <ExistingDocsMessage
                    setShowDialog={setShowDialog}
                    errors={titleError}
                    existingDocs={studyFiles}
                    fileNameInput={values.file_name}
                    newFile={!versionStatus}
                  />
                )}
              <Form onSubmit={handleSubmit} ref={ref}>
                <Form.Field required>
                  <label htmlFor="file_name">Document Title:</label>
                  <input
                    className={
                      touched.file_name && errors.file_name
                        ? 'border-red'
                        : null
                    }
                    data-testid="name-input"
                    type="text"
                    name="file_name"
                    placeholder="Phenotypic Data manifest for..."
                    value={values.file_name}
                    onBlur={handleBlur}
                    onChange={handleChange}
                  />
                  <small>
                    Please provide a descriptive title without dates or
                    adjectives such as "new", "updated", "final", etc.
                  </small>
                </Form.Field>
                {versionStatus && (
                  <Form.Field>
                    <label>Approval Status:</label>
                    <Dropdown
                      selection
                      fluid
                      name="file_status"
                      options={options}
                      value={values.file_status}
                      placeholder="Choose an option"
                      onChange={(e, {value}) => {
                        setFieldValue('file_status', value);
                      }}
                    />
                  </Form.Field>
                )}
                <Form.Field required>
                  <label htmlFor="file_type">Document Type:</label>
                  {Object.keys(fileTypeDetail).map(item => (
                    <Form.Field key={item}>
                      <Field
                        component={SelectElement}
                        name="file_type"
                        id={item}
                        label={item}
                      />
                    </Form.Field>
                  ))}
                </Form.Field>
                {submitButtons && (
                  <Form.Field required>
                    <label>Describe Document Contents:</label>
                    <Segment
                      className={
                        touched.file_desc && errors.file_desc
                          ? 'border-red'
                          : null
                      }
                    >
                      <MarkdownEditor
                        editorState={editorState}
                        onEditorStateChange={e => {
                          setEditorState(e);
                          validateForm();
                        }}
                      />
                    </Segment>
                  </Form.Field>
                )}
                {submitButtons &&
                  submitButtons(
                    Object.keys(errors).length > 0 ||
                      values.file_name.length === 0,
                    onUploading,
                  )}
              </Form>
            </>
          )}
        </Formik>
        {showDialog && (
          <UploadWizard
            startingStep={1}
            studyId={history.location.pathname.split('/')[2]}
            onCloseDialog={() => {
              setShowDialog(false);
            }}
            file={fileNode}
            fileList={studyFiles}
            history={history}
          />
        )}
      </>
    );
  },
);

EditDocumentForm.propTypes = {
  /** The title of the file */
  fileName: PropTypes.string,
  /** The description for the file */
  fileDescription: PropTypes.string,
  /** The value of the currently selected fileType's enum */
  fileType: PropTypes.string,
  /** The file approval status */
  versionStatus: PropTypes.string,
  /** If the user has ADMIN role */
  handleSubmit: PropTypes.func,
  /** (New file) Any errors that occured when submitting the form */
  errors: PropTypes.object,
  /** (New file) Displays as buttons for form submitting */
  submitButtons: PropTypes.func,
};

export default EditDocumentForm;
