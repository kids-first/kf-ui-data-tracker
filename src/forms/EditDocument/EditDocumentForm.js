/* eslint-disable  no-useless-escape */
import React, {useState} from 'react';
import PropTypes from 'prop-types';
import {Form, TextArea, Dropdown} from 'semantic-ui-react';
import {Formik, Field} from 'formik';
import validate from './validate';
import ExistingDocsMessage from './ExistingDocsMessage';
import DocumentTitleValidationIndicators from './DocumentTitleValidationIndicators';
import SelectElement from '../../components/FileAnnotation/SelectElement';
import Badge from '../../components/Badge/Badge';
import UploadWizard from '../../modals/UploadWizard/UploadWizard';
import {fileTypeDetail, versionState} from '../../common/fileUtils';

/**
 * Form to edit document information or annotate new file for uploading
 */
const EditDocumentForm = React.forwardRef(
  (
    {
      fileNode,
      fileType,
      fileName,
      fileDescription,
      versionStatus,
      isAdmin,
      history,
      handleSubmit,
      submitButtons,
      showFieldHints,
      studyFiles,
    },
    ref,
  ) => {
    const [onUploading, setUploading] = useState(false);
    const [showDialog, setShowDialog] = useState(false);
    const options = Object.keys(versionState).map(state => ({
      key: state,
      value: state,
      text: versionState[state].title,
      content: <Badge state={state} />,
    }));
    return (
      <>
        <Formik
          initialValues={{
            file_name: fileName,
            file_type: fileType,
            file_desc: fileDescription,
            file_status: versionStatus,
          }}
          validate={vals => validate(vals, fileNode, studyFiles)}
          onSubmit={values => {
            setUploading(true);
            handleSubmit(...Object.values(values));
          }}
        >
          {({
            values,
            handleChange,
            handleBlur,
            handleSubmit,
            setFieldValue,
            errors,
            touched,
          }) => (
            <>
              {showFieldHints &&
                errors.file_name &&
                (errors.file_name.existing_similarity ||
                  errors.file_name.upload_similarity) && (
                  <ExistingDocsMessage
                    setShowDialog={setShowDialog}
                    errors={errors}
                    existingDocs={studyFiles}
                    fileNameInput={values.file_name}
                  />
                )}
              <Form onSubmit={handleSubmit} ref={ref}>
                <Form.Field required>
                  <label htmlFor="file_name">Document Title:</label>
                  <Form.Input
                    data-testid="name-input"
                    type="text"
                    name="file_name"
                    placeholder="Phenotypic Data manifest for..."
                    value={values.file_name}
                    onBlur={handleBlur}
                    onChange={handleChange}
                    error={
                      errors.file_name &&
                      Object.values(errors.file_name).some(x => x != null)
                    }
                  />

                  <DocumentTitleValidationIndicators errors={errors} />
                </Form.Field>
                {versionStatus && (
                  <Form.Field>
                    <label>Approval Status:</label>
                    {isAdmin ? (
                      <Dropdown
                        selection
                        fluid
                        name="file_status"
                        options={options}
                        value={values.file_status}
                        placeholder="Choose an option"
                        onChange={(e, {value}) => {
                          setFieldValue('file_status', value);
                        }}
                      />
                    ) : (
                      <Badge state={versionStatus} />
                    )}
                  </Form.Field>
                )}

                <Form.Field required>
                  <label htmlFor="file_type">Document Type:</label>
                  {Object.keys(fileTypeDetail).map(item => (
                    <Form.Field key={item}>
                      <Field
                        component={SelectElement}
                        name="file_type"
                        id={item}
                        label={item}
                      />
                    </Form.Field>
                  ))}
                </Form.Field>

                <Form.Field required>
                  <label>Describe Document Contents:</label>
                  <TextArea
                    data-testid="description-input"
                    type="text"
                    name="file_desc"
                    value={values.file_desc}
                    onChange={handleChange}
                  />
                </Form.Field>
                {submitButtons &&
                  submitButtons(
                    Object.values(values).every(x => Boolean(x !== undefined)),
                    onUploading,
                  )}
              </Form>
            </>
          )}
        </Formik>
        {showDialog && (
          <UploadWizard
            startingStep={1}
            studyId={history.location.pathname.split('/')[2]}
            onCloseDialog={() => {
              setShowDialog(false);
            }}
            file={fileNode}
            fileList={studyFiles}
            history={history}
          />
        )}
      </>
    );
  },
);

EditDocumentForm.propTypes = {
  /** The title of the file */
  fileName: PropTypes.string,
  /** The description for the file */
  fileDescription: PropTypes.string,
  /** The value of the currently selected fileType's enum */
  fileType: PropTypes.string,
  /** The file approval status */
  versionStatus: PropTypes.string,
  /** If the user has ADMIN role */
  isAdmin: PropTypes.bool.isRequired,
  /** (New file) Function to perform on form submission  */
  handleSubmit: PropTypes.func,
  /** (New file) Any errors that occured when submitting the form */
  errors: PropTypes.string,
  /** (New file) Displays as buttons for form submitting */
  submitButtons: PropTypes.func,
  /** show validation hints */
  showFieldHints: PropTypes.Boolean,
};

export default EditDocumentForm;
