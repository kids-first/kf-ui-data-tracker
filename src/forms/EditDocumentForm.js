import React from 'react';
import PropTypes from 'prop-types';
import classes from 'classnames';
import {GridContainer, Icon} from 'kf-uikit';
import SelectElement from '../components/FileAnnotation/SelectElement';
import Badge from '../components/Badge/Badge';
import {fileTypeDetail, versionState} from '../common/fileUtils';
/**
 * Form to edit document information
 */
const EditDocumentForm = ({
  fileType,
  fileName,
  fileDescription,
  versionStatus,
  onFileTypeChange,
  onNameChange,
  onDescriptionChange,
  onVersionStatusChange,
}) => {
  const versionStatusList = Object.keys(versionState);
  const dropdownClass = classes('Dropdown');
  return (
    <form className="pt-16">
      <GridContainer>
        <label
          className="Form--Label row-1 cell-3 text-right Form--Label-required"
          htmlFor="file_name"
        >
          Document Title:
        </label>
        <input
          data-testid="name-input"
          className="Form--Input border border-grey row-1 cell-9"
          type="text"
          name="file_name"
          placeholder="Phenotypic Data manifest for..."
          defaultValue={fileName}
          onChange={e => onNameChange(e)}
        />
        <label
          className="Form--Label row-2 cell-3 text-right"
          htmlFor="file_name"
        >
          Approval Status:
        </label>
        <div className="row-2 cell-9">
          <div className={dropdownClass}>
            <button
              type="button"
              className="Dropdown--button justify-start pl-20"
              tabIndex="0"
              data-testid="status-dropdown"
            >
              <Badge state={versionStatus} />
              <Icon
                className="m-12"
                width={20}
                height={20}
                kind="chevron-down"
              />
            </button>
            <div className="Dropdown--container">
              <ul className="Dropdown--list">
                {versionStatusList.map(versionStatusValue => (
                  <button
                    onClick={() => onVersionStatusChange(versionStatusValue)}
                    type="button"
                    className="Dropdown--item"
                    tabIndex="0"
                    key={versionStatusValue}
                  >
                    <Badge state={versionStatusValue} />
                  </button>
                ))}
              </ul>
            </div>
          </div>
        </div>
        <label
          className="Form--Label row-3 cell-3 text-right Form--Label-required"
          htmlFor="file_type"
        >
          Document Type:
        </label>
        <fieldset name="file_type" className="row-3 cell-9">
          {Object.keys(fileTypeDetail).map(item => (
            <SelectElement
              key={item}
              value={item}
              select={e => onFileTypeChange(e)}
              selected={fileType === item}
            />
          ))}
        </fieldset>
        <label
          className="Form--Label row-4 cell-3 text-right Form--Label-required"
          htmlFor="description"
        >
          Describe document contents:
        </label>
        <textarea
          data-testid="description-input"
          className="Form--TextArea border border-grey w-full row-4 cell-9"
          type="text"
          name="description"
          defaultValue={fileDescription}
          onChange={e => onDescriptionChange(e)}
        />
      </GridContainer>
    </form>
  );
};

EditDocumentForm.propTypes = {
  /** The title of the file */
  fileName: PropTypes.string,
  /** The description for the file */
  fileDescription: PropTypes.string,
  /** The value of the currently selected fileType's enum*/
  fileType: PropTypes.string,
  /** Action to perform when name input is updated */
  onNameChange: PropTypes.func.isRequired,
  /** Action to perform when description input is updated */
  onDescriptionChange: PropTypes.func.isRequired,
  /** Action to perform when fileType input is updated */
  onFileTypeChange: PropTypes.func.isRequired,
};

export default EditDocumentForm;
