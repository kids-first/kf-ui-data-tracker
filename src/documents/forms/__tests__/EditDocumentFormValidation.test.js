/* eslint-disable no-loop-func */
import React from 'react';
import wait from 'waait';
import {MockedProvider} from '@apollo/react-testing';
import {MemoryRouter} from 'react-router-dom';
import {mocks} from '../../../../__mocks__/kf-api-study-creator/mocks';
import studyByKfId from '../../../../__mocks__/kf-api-study-creator/responses/studyByKfId.json';
import {Segment, Button} from 'semantic-ui-react';
import {render, fireEvent, cleanup, act} from 'react-testing-library';
import EditDocumentForm from '../EditDocumentForm';

afterEach(cleanup);

let file = {
  lastModified: 1567797130510,
  name: 'organize.tsv',
  size: 7,
  type: 'text/tab-separated-values',
  webkitRelativePath: '',
};

const historyMock = {
  push: jest.fn(),
  location: {pathname: '/study/SD_8WX8QQ06/documents/new-documents'},
};
let errors = {};

it('Show warning message when input file name is similar to an existing file', async () => {
  const handleSubmit = jest.fn();

  const {container, getByTestId, getByText, queryAllByText} = render(
    <MockedProvider mocks={mocks}>
      <MemoryRouter
        initialEntries={['/study/SD_8WX8QQ06/documents/new-document']}
      >
        <EditDocumentForm
          studyFiles={studyByKfId.data.studyByKfId.files.edges}
          isAdmin={false}
          fileNode={file}
          handleSubmit={handleSubmit}
          errors={errors}
          history={historyMock}
          showFieldHints={true}
          submitButtons={(disabled, onUploading) => (
            <Segment vertical basic compact>
              <Button
                floated="right"
                type="submit"
                primary={errors.length === 0}
                disabled={disabled}
              >
                {onUploading && !errors ? 'UPLOADING ...' : 'UPLOAD'}
              </Button>
              <Button floated="right" primary={errors.length > 0}>
                CANCEL
              </Button>
            </Segment>
          )}
        />
      </MemoryRouter>
    </MockedProvider>,
  );

  await wait(0);

  //react portal w/ modal is added to document
  expect(document.querySelectorAll('.ui.form').length).toBe(1);
  expect(container).toMatchSnapshot();

  // Update file name with an existing file
  // this should make the form invalid and block submission
  const nameInput = getByTestId('name-input');
  act(() => {
    fireEvent.change(nameInput, {
      target: {value: studyByKfId.data.studyByKfId.files.edges[0].node.name},
    });
  });
  await wait();
  expect(container).toMatchSnapshot();

  expect(queryAllByText(/UPLOAD/).length).toBe(1);

  act(() => {
    fireEvent.click(getByText(/Update Existing Document/i));
  });

  await wait();
  expect(container).toMatchSnapshot();
});
