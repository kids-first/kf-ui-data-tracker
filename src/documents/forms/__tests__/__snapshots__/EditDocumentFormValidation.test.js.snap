// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blocks submission for invalid Document Name input 1`] = `
<div>
  <form
    class="ui form"
  >
    <div
      class="required field"
    >
      <label
        for="file_name"
      >
        Document Title:
      </label>
      <input
        data-testid="name-input"
        error="undefined"
        name="file_name"
        placeholder="Phenotypic Data manifest for..."
        type="text"
        value=""
      />
    </div>
    <div
      class="required field"
    >
      <label
        for="file_type"
      >
        Document Type:
      </label>
      <div
        class="field"
      >
        <div
          class="ui compact segment selectionRadio--card"
        >
          <div>
            <div
              class="ui radio checkbox"
            >
              <input
                class="hidden"
                id="SHM"
                name="file_type"
                readonly=""
                tabindex="0"
                type="radio"
                value="SHM"
              />
              <i
                aria-hidden="true"
                class="black shipping large bordered circular inverted icon"
              />
            </div>
          </div>
          <div>
            <p
              class="mb-5"
            >
              <b>
                Biospecimen Manifest
              </b>
            </p>
            <p
              class="text-wrap-75 text-grey"
            >
              Tabular files (e.g. CSV, TSV, Excel) containing biospecimen identifiers and metadata (e.g. tissue type, composition) provided to sequencing centers as well as consent type.
            </p>
          </div>
        </div>
      </div>
      <div
        class="field"
      >
        <div
          class="ui compact segment selectionRadio--card"
        >
          <div>
            <div
              class="ui radio checkbox"
            >
              <input
                class="hidden"
                id="CLN"
                name="file_type"
                readonly=""
                tabindex="0"
                type="radio"
                value="CLN"
              />
              <i
                aria-hidden="true"
                class="black hospital large bordered circular inverted icon"
              />
            </div>
          </div>
          <div>
            <p
              class="mb-5"
            >
              <b>
                Clinical/Phenotypic Data
              </b>
            </p>
            <p
              class="text-wrap-75 text-grey"
            >
              Structured files with data linked to the biospecimen identifiers provided in the shipping manifest(s). Exports from data management applications, such as REDCap, are the fastest to process, but we will work with you on most formats.
            </p>
          </div>
        </div>
      </div>
      <div
        class="field"
      >
        <div
          class="ui compact segment selectionRadio--card"
        >
          <div>
            <div
              class="ui radio checkbox"
            >
              <input
                class="hidden"
                id="SEQ"
                name="file_type"
                readonly=""
                tabindex="0"
                type="radio"
                value="SEQ"
              />
              <i
                aria-hidden="true"
                class="black dna large bordered circular inverted icon"
              />
            </div>
          </div>
          <div>
            <p
              class="mb-5"
            >
              <b>
                Sequencing Manifest
              </b>
            </p>
            <p
              class="text-wrap-75 text-grey"
            >
              Tabular files (e.g. CSV, TSV, Excel) containing the biospecimen identifiers and the associated genomic data files. Typically provided by the sequencing center.
            </p>
          </div>
        </div>
      </div>
      <div
        class="field"
      >
        <div
          class="ui compact segment selectionRadio--card"
        >
          <div>
            <div
              class="ui radio checkbox"
            >
              <input
                class="hidden"
                id="OTH"
                name="file_type"
                readonly=""
                tabindex="0"
                type="radio"
                value="OTH"
              />
              <i
                aria-hidden="true"
                class="black question large bordered circular inverted icon"
              />
            </div>
          </div>
          <div>
            <p
              class="mb-5"
            >
              <b>
                Other
              </b>
            </p>
            <p
              class="text-wrap-75 text-grey"
            >
              Any useful documents, such as study background information, data dictionaries, etc., that does not clearly fit in the other categories.
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="required field"
    >
      <label>
        Describe Document Contents:
      </label>
      <textarea
        data-testid="description-input"
        name="file_desc"
        rows="3"
        type="text"
      />
    </div>
    <div
      class="ui basic compact vertical segment"
    >
      <button
        class="ui right floated button"
        type="submit"
      >
        UPLOAD
      </button>
      <button
        class="ui right floated button"
      >
        CANCEL
      </button>
    </div>
  </form>
</div>
`;
